compose.py

```diff
@@ -0,0 +1,280 @@
+            'docker inspect of upstream docker image build sha failed')
+    p_build = subprocess.run(baseRunArgs + [
+        '{{ index .Config.Labels "com.nvidia.build.id"}}', upstreamDockerImage
+    ],
+                             capture_output=True,
+                             text=True)
+    fail_if(p_build.returncode != 0,
+            'docker inspect of upstream docker image build sha failed')
+
+    p_find = subprocess.run(
+        ['docker', 'run', upstreamDockerImage, 'bash', '-c', 'ls /usr/bin/'],
+        capture_output=True,
+        text=True)
+    f = re.search("serve", p_find.stdout)
+    fail_if(p_find.returncode != 0, "Cannot search for 'serve' in /usr/bin")
+    argmap = {
+        'NVIDIA_BUILD_REF': p_sha.stdout.rstrip(),
+        'NVIDIA_BUILD_ID': p_build.stdout.rstrip(),
+        'TRITON_VERSION': version,
+        'TRITON_CONTAINER_VERSION': container_version,
+        'ENDPOINTS': f is not None,
+    }
+    return argmap
+
+
+# Install dependencies and run entrypoint script
+def end_gpu_dockerfile(ddir, dockerfile_name, argmap, backends):
+    import build
+    df = build.dockerfile_add_installation_linux(argmap, backends)
+    if argmap['ENDPOINTS']:
+        df += '''
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+COPY --chown=1000:1000 --from=full /usr/bin/serve /usr/bin/.
+'''
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+### Create container with docker build
+def build_docker_image(ddir, dockerfile_name, container_name):
+    p = subprocess.Popen(['docker', 'build', '-t', container_name, '-f', \
+        os.path.join(ddir, dockerfile_name), '.'])
+    p.wait()
+    fail_if(p.returncode != 0, 'docker build {} failed'.format(container_name))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    group_qv = parser.add_mutually_exclusive_group()
+    group_qv.add_argument('-q',
+                          '--quiet',
+                          action="store_true",
+                          required=False,
+                          help='Disable console output.')
+    group_qv.add_argument('-v',
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument(
+        '--output-name',
+        type=str,
+        required=False,
+        help='Name for the generated Docker image. Default is "tritonserver".')
+    parser.add_argument(
+        '--work-dir',
+        type=str,
+        required=False,
+        help=
+        'Generated dockerfiles are placed here. Default to current directory.')
+    parser.add_argument(
+        '--container-version',
+        type=str,
+        required=False,
+        help=
+        'The version to use for the generated Docker image. If not specified the container version will be chosen automatically based on --version value.'
```

Comment:
There is no --version flag, so change desc to be "... will be chosen automatically based on the repository branch."